db/system_keyspace.cc

```diff
@@ -1763,7 +1764,8 @@ future<> system_keyspace::drop_truncation_rp_records() {
     auto rs = co_await execute_cql(req);
 
     bool any = false;
-    co_await coroutine::parallel_for_each(*rs, [&] (const cql3::untyped_result_set_row& row) -> future<> {
+    auto max_concurrency = std::max(1024u, smp::count * 8);
```

Comment:
Sorry, duh, I meant std::min, not max.
It won't make much difference with the typical number of shards we run on, but the top level tasks will run on shard 0, hence the 1024 cap.

Reply:
Makes more sense. :-)